src/qml/main.qml

```diff
@@ -231,7 +231,26 @@ Kirigami.ApplicationWindow {
     }
     Component {
         id: welcomePage;
-        WelcomePage {}
+
+        WelcomePage {
+            onBackRequested: {
+                if (!BTConnectionManager.isConnected) {
+                    return;
+                }
+
+                event.accepted = true;
+
+                showMessageBox(qsTr("A tail is currently connected"),
+                               qsTr("A tail is currently connected.\n\nAre you sure that you want to disconnect from the tail and quit?"),
```

Comment:
We probably want to just have this not duplicate the title in the message, but we can look at that later, the functionality's solid :)